libcudacxx/include/cuda/std/__bit/byteswap.h

```diff
@@ -87,13 +97,16 @@ class __byteswap_impl
 #if defined(_CCCL_BUILTIN_BSWAP16)
     return _CCCL_BUILTIN_BSWAP16(__val);
 #else // ^^^ _CCCL_BUILTIN_BSWAP16 ^^^ / vvv !_CCCL_BUILTIN_BSWAP16 vvv
-#  if _CCCL_COMPILER(MSVC)
     if (!_CUDA_VSTD::__cccl_default_is_constant_evaluated())
     {
+#  if _CCCL_COMPILER(MSVC)
       NV_IF_TARGET(NV_IS_HOST, return _byteswap_ushort(__val);)
-    }
 #  endif // _CCCL_COMPILER(MSVC)
-    return (__val << CHAR_BIT) | (__val >> CHAR_BIT);
+#  if __cccl_ptx_isa >= 200
+      NV_IF_TARGET(NV_PROVIDES_SM_50, return __impl_device(__val);)
+#  endif // __cccl_ptx_isa >= 200
```

Comment:
We only support architectures above 50, so I would just replace that with `NV_IS_DEVICE`